.github/workflows/e2e.yaml

```diff
@@ -13,19 +13,30 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Check out code
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: |
+            limitador-testing
+          tags: |
+            type=raw,value=latest
       - name: Build Limitador Image
         id: build-image
-        uses: redhat-actions/buildah-build@v2
+        uses: docker/build-push-action@v5
```

Comment:
Similar changes were **not** done to https://github.com/Kuadrant/limitador/blob/38f55ca1c96246b6532351207453efbf63438c33/.github/workflows/build-image.yaml#L2 due to issues mentioned by https://github.com/Kuadrant/limitador/pull/215.

Additionally as we pass the `GITHUB_SHA` build arg to the dockerfile, we always compile and build on each push as the SHA is always different, so we won't pull from cache which is the most time intensive layer
https://github.com/Kuadrant/limitador/blob/38f55ca1c96246b6532351207453efbf63438c33/.github/workflows/build-image.yaml#L42-L43
https://github.com/Kuadrant/limitador/blob/38f55ca1c96246b6532351207453efbf63438c33/Dockerfile#L26